### PR TITLE
[sui-execution] Move `sui` execution crates into their own directory

### PR DESCRIPTION
**Pull request by amnn**
## Description

Create the sub-directory containing the "latest" version of `sui-` crates in the execution layer.  API to access multiple versions of an execution crate to follow.  This commit just moves code to its new home, to minimise the chance for a merge conflict.

## Test Plan

```
$ cargo simtest
$ env SUI_SKIP_SIMTESTS=1 cargo nextest run
```

### CODE REVIEW

**Comment by amnn** [comment]
Addressing some potential review comments up-front:

- Originally, `sui-execution` was within `crates` but this triggered a linter warning that `crates` needs to have a flat directory structure.
- It's useful to have all the execution crates from the same version in the same directory (e.g. `sui-execution/latest`) rather than flatten them out, because then they can refer to each other by relative paths, and when we cut a new version, we don't need to touch this part.

**Comment by dariorussi** [comment]
I see what we are doing... this is so much more clear that the simple contrived example you pointed me to :D 
thanks

**Review by tnowacki — APPROVED** [review]

File: crates/sui-config/Cargo.toml [hunk 1]
```diff
@@ -25,12 +25,14 @@ tempfile = "3.3.0"
 tracing = "0.1.36"
 prometheus = "0.13.3"
 
+# TODO use sui-execution versioned crate
```
Comment (dariorussi) [inline crates/sui-config/Cargo.toml:28]: what is this TODO doing or saying?

File: crates/sui-core/Cargo.toml [hunk 2]
```diff
@@ -55,6 +55,10 @@ telemetry-subscribers.workspace = true
 typed-store-derive.workspace = true
 typed-store.workspace = true
 
+# TODO use sui-execution versioned crate
```
Comment (dariorussi) [inline crates/sui-core/Cargo.toml:58]: TODO?

File: crates/sui-core/Cargo.toml [hunk 3]
```diff
@@ -91,9 +93,11 @@ serde_yaml = "0.8.26"
 
 move-symbol-pool.workspace = true
 
+# TODO use sui-execution versioned crate
```
Comment (dariorussi) [inline crates/sui-core/Cargo.toml:96]: TODO?

File: crates/sui-replay/Cargo.toml [hunk 4]
```diff
@@ -30,13 +30,15 @@ move-package.workspace = true
 tokio.workspace = true
 typed-store.workspace = true
 
-sui-adapter = { path = "../sui-adapter" }
+# TODO use sui-execution versioned crate
```
Comment (dariorussi) [inline crates/sui-replay/Cargo.toml:33]: TODO?

**Review by dariorussi — APPROVED** [review]
left few comments about TODO I did not understand.
What about the MoveVM? are we going to create a versioned story for that too?
Am I correct that there is nothing for it right now?
I understand that by versioning the execution layer you can effectively version the VM as well but wondering....
thoughts?

**Comment by amnn** [comment]
@dariorussi, versioning for the move crates and the `sui-execution` versioned crate that the TODOs refer to are to come in a follow-up PR, because they are also pretty big changes on their own.  The TODOs are meant to be here as a reminder to me for where I need to change references to the "latest" version of the execution crate to point to the crate that dispatches by version.

**Pull request merged**